### PR TITLE
UIX-15, UIX-4: Added basic API wrapper, added event management wrapper

### PR DESCRIPTION
**Pull request by psy-tama**
**------ For PR creator ------**

UIX-15: Added basic API wrapper
UIX-4: Added event management wrapper

## Creator Checklist

- [x] Self review
- [ ] If this is a bug, add small root cause and fix
- [ ] If UI change, attach screenshot or recording

**-------- For Reviewer ------**

## Reviewer Checklist

- [ ] JIRA is linked correctly
- [ ] Base branch is main

### CODE REVIEW

**Comment by psy-tama** [comment]
@bjorn-uix @takashi-uix Added a basic version of API wrapper with useSWR, types are not used properly as it s not configured completely.

**Comment by psy-tama** [comment]
This is now fully typed and functional.

File: src/hooks/useFetch/useFetch.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  url,
+  options,
+  configOverride = {},
+  enabled = true
+}: FetchArgTypes): SWRResponse<T, Error> => {
+  const fetcher: Fetcher<T, Key> = async (
+    url: string,
+    options?: RequestInit
+  ): Promise<T> => {
+    const res = await fetch(url, options);
```
Comment (takashi-uix) [inline src/hooks/useFetch/useFetch.ts:21]: How will we inject token if required..?

File: src/hooks/useFetch/helper.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const sanitizeOptions = (options?: any): any => {
+  const requestBody = options?.body;
+  if (requestBody) {
+    return {
+      ...options,
+      body: JSON.stringify(requestBody)
```
Comment (takashi-uix) [inline src/hooks/useFetch/helper.ts:6]: any other way to do this?

File: src/hooks/useFetch/useFetch.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    return res.json();
+  };
+
+  const config = {
+    shouldRetryOnError: false
+  };
+  const sanitizedOptions = sanitizeOptions(options);
+  return useSWR<T, Error>(
+    () => (enabled ? [url, sanitizedOptions] : null),
```
Comment (takashi-uix) [inline src/hooks/useFetch/useFetch.ts:38]: if not enabled nothing will happen i see no logic when enabled === false ?

**Review by takashi-uix — CHANGES_REQUESTED** [review]

File: src/hooks/useFetch/useFetch.ts (hunk 1 above)
Comment (psy-tama) [reply src/hooks/useFetch/useFetch.ts:21]: `options` is the option part or config part of a fetch request, which consists of method, body, header etc. we can modify as per our requirements. We can even add some utility functions based on our requirement if we want to forward as cookie or get it from session storage, but that depends on the project

File: src/hooks/useFetch/helper.ts (hunk 2 above)
Comment (psy-tama) [reply src/hooks/useFetch/helper.ts:6]: another way to do is stringify it when using `useFetch` but I think this is more convenient, as this function always stringifies the body which is required by fetch

File: src/hooks/useFetch/useFetch.ts (hunk 3 above)
Comment (psy-tama) [reply src/hooks/useFetch/useFetch.ts:38]: enabled is the trigger of a particular useFetch hook, if we want to disable the fetch for a certain case we can do that by sending `enabled={false}`, if the first argument of `useSwr` gets a falsy value it does not fetch.

Comment (takashi-uix) [reply src/hooks/useFetch/useFetch.ts:38]: tested both flows?

Comment (takashi-uix) [reply src/hooks/useFetch/useFetch.ts:38]: can we also add comments to as what each option does?

File: src/hooks/useFetch/helper.ts (hunk 2 above)
Comment (takashi-uix) [reply src/hooks/useFetch/helper.ts:6]: Please add a comment on how to add token

File: src/hooks/useFetch/useFetch.ts (hunk 3 above)
Comment (psy-tama) [reply src/hooks/useFetch/useFetch.ts:38]: will add a detailed document with each use case next.

**Comment by psy-tama** [comment]
@takashi-uix Added event management wrapper also

**Review by takashi-uix — APPROVED** [review]

**Pull request merged**